### PR TITLE
[DDW-759] Over-saturation warning

### PR DESCRIPTION
**Pull request by mchappell**
This PR adds  warning message in the delegation wizard to warn users if over-saturation will occur when delegation to a pool.

## Todos

- [x] Create component
- [x] Update delegation wizard to check if over-saturation will occur
- [x] Update storybook

## Screenshots
Screenshot 2021-11-09 at 10 40 15
Screenshot 2021-11-09 at 10 40 21
Screenshot 2021-11-09 at 10 40 37
Screenshot 2021-11-09 at 10 40 40

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

## Test Scenario

```Gherkin
Scenario:Delegate wallet that results in oversaturated pool
Given I have a wallet with ADA
And there is a pool which is nearly saturated or oversaturated
And if I delegate the wallet to the pool, the pool will become oversaturated
When I delegate the wallet to the pool
Then at CHOOSE A STAKE POOL - STEP 2 OF 3
And at CONFIRM DELEGATION - STEP 3 OF 3
I will observe this “warning_message” (Note: % will differ depending on the oversaturation level)

warning_message
The selected stake pool will become oversaturated by 115%, which will reduce future rewards for all delegators to that pool.
選択したステークプールに委任すると飽和率が115%になり、このプールへのすべての委任者が今後受け取る報酬が減少します。
```

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by gabriela-ponce** [comment]
@mchappell Here's a list of the issues @dmitrii-gaico and I found on 19556:

- When the pool is already delegated to an oversaturated pool, you can still see the error on step 2. Check [this](https://input-output-rnd.slack.com/files/U02195E94PJ/F02L4MXA0CW/delegate.mp4) video.
- The messages for retiring pools and oversaturation look different. Check [this](https://input-output-rnd.slack.com/files/UK8QCC2N6/F02LQ64PRR7/nov-05-2021_11-16-36.mp4) video.
- Fee calculation takes longer (or doesn't calculate at all) when you delegate from the stake pools screen (instead of delegation center). Works as expected on production and develop. Check [this](https://drive.google.com/file/d/1SMOwbV63B8KrDa953klAhpuxiu3yFJLa/view?usp=sharing) video.

**Comment by mchappell** [comment]
> @mchappell Here's a list of the issues @dmitrii-gaico and I found on 19556:
> 
> * When the pool is already delegated to an oversaturated pool, you can still see the error on step 2. Check [this](https://input-output-rnd.slack.com/files/U02195E94PJ/F02L4MXA0CW/delegate.mp4) video.
> * The messages for retiring pools and oversaturation look different. Check [this](https://input-output-rnd.slack.com/files/UK8QCC2N6/F02LQ64PRR7/nov-05-2021_11-16-36.mp4) video.
> * Fee calculation takes longer (or doesn't calculate at all) when you delegate from the stake pools screen (instead of delegation center). Works as expected on production and develop. Check [this](https://drive.google.com/file/d/1SMOwbV63B8KrDa953klAhpuxiu3yFJLa/view?usp=sharing) video.

@gabriela-ponce
- Already delegated pools now ignored from over-saturation calculation
- Existing font choice for retiring message (incorrect) now updated to match designs
- Fee calculations fixed for both from delegation and stake pools list

**Review by gabriela-ponce — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Tested on 19570. Great work @mchappell 👍

**Review by ManusMcCole — COMMENTED** [review]
Good work @mchappell  👍

**Review by tomislavhoracek — APPROVED** [review]
@mchappell 💯 🚀

**Review by dmitrii-gaico — APPROVED** [review]

File: source/renderer/app/components/staking/delegation-setup-wizard/DelegationSetupWizardDialog.js [hunk 1]
```diff
@@ -45,6 +45,25 @@ type Props = {
...
 
+const getOversaturationPercentage = (
+  selectedWallet: ?Wallet,
+  selectedPool: ?StakePool,
+  maxDelegationFunds: number
+): number => {
+  if (
+    !selectedPool ||
+    !selectedWallet ||
+    (selectedWallet.delegatedStakePoolId ||
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-setup-wizard/DelegationSetupWizardDialog.js:56]: @mchappell are we sure this is the right logic?

e.g. we have a case: `StakePoolA` is the `selectedPool`:

```
selectedWallet.delegatedStakePoolId = StakePoolB
selectedWallet.lastDelegatedStakePoolId = StakePoolA
```

☝️  this case would result in **not showing** the oversaturation notification which is fine 


```
selectedWallet.delegatedStakePoolId = StakePoolA
selectedWallet.lastDelegatedStakePoolId = StakePoolB
```

☝️  this case would result in **not showing** the oversaturation notification and it should!

File: source/renderer/app/components/staking/delegation-setup-wizard/DelegationStepsConfirmationDialog.js [hunk 2]
```diff
@@ -272,7 +200,11 @@ export default class DelegationStepsConfirmationDialog extends Component<Props>
             labelDisabled
           />
         </div>
-
+        {oversaturationPercentage > 0 && (
+          <OversaturationText
+            oversaturationPercentage={oversaturationPercentage.toFixed()}
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/delegation-setup-wizard/DelegationStepsConfirmationDialog.js:205]: @mchappell shouldn't we have 2 decimal places here?

**Review by nikolaglumac — APPROVED** [review]
Great work @mchappell 🎉

**Pull request merged**